contracts/StablecoinConverter.sol

```diff
@@ -249,18 +249,20 @@ contract StablecoinConverter is EpochTokenLocker {
         }
     }
 
-    function getExecutedBuyAmount(
-        uint128 executedSellAmount,
+    function getExecutedSellAmount(
+        uint128 executedBuyAmount,
         uint128 buyTokenPrice,
         uint128 sellTokenPrice
     ) internal view returns (uint128) {
-        uint128 buyAmount = uint128(executedSellAmount.mul(sellTokenPrice) / buyTokenPrice);
-        // executedBuyAmount = buyAmount * (1 - (1/feeDenominator))
-        //                   = buyAmount - buyAmount/feeDenominator (*)
-        //                   = (buyAmount * feeDenominator)/ feeDenominator - buyAmount/feeDenominator
-        //                   = (buyAmount * feeDenominator - buyAmount) / feeDenominator
-        //                   = (buyAmount * (feeDenominator - 1)/feeDenominator
-        return uint128(buyAmount.mul(feeDenominator - 1) / feeDenominator);
+        uint128 sellAmount = uint128(executedBuyAmount.mul(buyTokenPrice) / sellTokenPrice);
+        // executedSellAmount * p[sellToken] * (1-phi) == executedBuyAmount * p[buyToken]
+        // phi = 1 / feeDenominator (since we only have integers)
+        // executedSellAmount = (executedBuyAmount * p[buyToken] / p[sellToken]) * (1 / (1 - phi))
+        // executedSellAmount = sellAmount * (1 / (1 - phi))
+        // executedSellAmount = sellAmount * (1 / (1 - 1 / feeDenominator))
+        // executedSellAmount = sellAmount * (feeDenominator / (feeDenominator - 1))
+        //                    = (sellAmount * feeDenominator) / (feeDenominator - 1)
+        return uint128(sellAmount.mul(feeDenominator) / (feeDenominator - 1));
```

Comment:
Didn't we say we wanted to first divide by (feeDenominator - 1), then multiply by feeDenominator and then divide by sellTokenPrice to keep the rounding error small?

Cf. https://github.com/gnosis/dex-services/pull/252/commits/1e34de88681e56781b9a26b96ddba7a2b62817f3#diff-6bd9f48fcec6a68cb1ee81d59a749238

Reply:
Yes, sorry. I am going to close this in favour of a branch @josojo  has already created [here](https://github.com/gnosis/dex-contracts/tree/buyAmounts)